binding_generator.py

```diff
@@ -168,6 +168,7 @@ def generate_bindings(api_filepath, use_template_get_node, bits="64", double="fl
     shutil.rmtree(target_dir, ignore_errors=True)
     target_dir.mkdir(parents=True)
 
+    double = "double" if precision == "double" else "float"
```

Comment:
`double` as a variable name here seems wrong.

I'd consider just modifying the `precision` variable, and using that later:

```py
if precision == "single":
    precision = "float"
```

Reply:
I inherited the variable name from the parameter name before this PR, but I'll update it to `real_t`. The `FLOAT_PRECISION_FLAG` part of the patch above is not part of this PR because I restructured it.